### ISSUE TITLE
Release?

### ISSUE THREAD

**Issue by rabernat** [xgcm/xgcm#217]
And are also in the process of fixing some bugs (#215, #210). We have also had a few features (#186, #152) since the 0.3.0 release in February. Do we want to do another release? Are these new features major enough to do 0.4.0? Or are we talking about 0.3.1?

Let's try to do a release this week.

**Comment by jbusecke** [comment]
Good idea. Id opt for 0.3.1. I would do that right after I fix #215.

**Comment by jbusecke** [comment]
We should def try to release more frequently. How do other projects manage this? Just do a weekly release?

**Comment by rabernat** [comment]
Our release frequency should match our development rate. There's not point in releasing if there are no new features / bug fixes. I'd like to streamline the release process as much as possible to make it easy to release whenever we have anything new.

**Comment by jbusecke** [comment]
Yea, you are right, I just want to realease more often if we fix bugs. I think at this point it is quite streamlined, but I will watch out for any hickups in the next minor release.

**Comment by jbusecke** [comment]
Il make some changes to the `whats-new.rst` and release as 0.4.0.

**Comment by jbusecke** [comment]
I seem to have some [problems](https://github.com/xgcm/xgcm/actions/runs/236795478) with the authentication in our publishing gh action. 

The logs show this:

```
2020-09-02T22:07:51.9769641Z removing build/bdist.linux-x86_64/wheel
2020-09-02T22:07:52.7011864Z Uploading distributions to https://upload.pypi.org/legacy/
2020-09-02T22:07:52.7013568Z Uploading xgcm-0.4.0-py3-none-any.whl
2020-09-02T22:07:52.7014237Z 
2020-09-02T22:07:52.8054298Z   0%|          | 0.00/57.2k [00:00<?, ?B/s]
2020-09-02T22:07:52.9504799Z  14%|█▍        | 8.00k/57.2k [00:00<00:00, 78.8kB/s]
2020-09-02T22:07:52.9519476Z 100%|██████████| 57.2k/57.2k [00:00<00:00, 235kB/s] 
2020-09-02T22:07:52.9519897Z NOTE: Try --verbose to see response content.
2020-09-02T22:07:52.9524699Z HTTPError: 403 Forbidden from https://upload.pypi.org/legacy/
2020-09-02T22:07:52.9525394Z Invalid or non-existent authentication information. See https://pypi.org/help/#invalid-auth for more information.
2020-09-02T22:07:52.9845380Z ##[error]Process completed with exit code 1.
2020-09-02T22:07:52.9865715Z Cleaning up orphan processes
```
@rabernat could you check on the pypi authentication details as suggested [here](https://pypi.org/help/#invalid-auth)?

**Comment by rabernat** [comment]
I found we had both repo secrets and organization secrets for pypi.

![image](https://user-images.githubusercontent.com/1197350/92059473-4f84ca00-ed5f-11ea-9f6d-1f44b7383be7.png)

I deleted the repo secrets. We will use the org secrets. Let's see if that fixes it.

**Comment by rabernat** [comment]
Looks like 0.4.0 is released! https://pypi.org/project/xgcm/

✋

**Comment by jbusecke** [comment]
🔥

**Comment by rabernat** [comment]
It looks like the conda build for 0.4.0 [failed](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=204382&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d) with the same issue described in #210. I think we probably have to update the [conda dependencies](https://github.com/conda-forge/xgcm-feedstock/blob/master/recipe/meta.yaml) as well.

**Comment by jbusecke** [comment]
Ok Ill change that. It would be cool if this could be taken automatically from the setup.py, though. Do you know if there is a way to do that?

**Comment by jbusecke** [comment]
Fixed the dependencies and the tests pass. Once the new version is uploaded. Ill download and test it and close this and #210

**Comment by jbusecke** [comment]
Somehow my conda still resolves to 0.3.0. Ill give it some time and try again in a bit.

**Comment by jbusecke** [comment]
Just installed and tested 0.4.0 in a pristine environment. Closing this now.
